doc/man7/EVP_PKEY-DH.pod

```diff
@@ -41,31 +47,45 @@ The following values can be used by the OpenSSL's default and FIPS providers:
 The following additional values can also be used by OpenSSL's default provider:
 "modp_1536", "dh_1024_160", "dh_2048_224", "dh_2048_256".
 
-DH named groups can be easily validated since the parameters are well known.
+DH/DHX named groups can be easily validated since the parameters are well known.
 For protocols that only transfer I<p> and I<g> the value of I<q> can also be
 retrieved.
 
-=item "safeprime-generator" (B<OSSL_PKEY_PARAM_DH_GENERATOR>) <integer>
+=back
 
-Used for DH generation of safe primes using the old generator code.
-It is recommended to use a named safe prime group instead, if domain parameter
-validation is required. The default value is 2.
+=head2 DH and DHX additional parameters
 
-These are not named safe prime groups so setting this value for the OpenSSL FIPS
-provider will instead choose a named safe prime group based on the size of I<p>.
+=over 4
 
 =item "encoded-pub-key" (B<OSSL_PKEY_PARAM_ENCODED_PUBLIC_KEY>) <octet string>
 
 Used for getting and setting the encoding of the DH public key used in a key
 exchange message for the TLS protocol.
+See EVP_PKEY_set1_encoded_public_key() and EVP_PKEY_get1_encoded_public_key().
+
+=back
+
+=head2 DH additional domain parameters
+
+=over 4
+
+=item "safeprime-generator" (B<OSSL_PKEY_PARAM_DH_GENERATOR>) <integer>
+
+Used for DH generation of safe primes using the old safe prime generator code.
+The default value is 2.
+It is recommended to use a named safe prime group instead, if domain parameter
+validation is required. 
+
+These are not 'named' safe prime groups so setting this value for the OpenSSL FIPS
```

Comment:
Maybe: `Randomly generated safe prime groups are not allowed by FIPS so setting ...`